nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestCommentNotification.java

```diff
@@ -149,12 +162,90 @@ public void shouldNotifyWithTheRightCommentedDocument() {
 
         // Reply
         captureAndVerifyCommentEventNotification(() -> {
-            Comment reply = createComment(commentedDocumentModel);
+            Comment reply = createComment(commentedDocumentModel, ADMINISTRATOR, ANY_COMMENT_MESSAGE);
             DocumentModel replyDocumentModel = session.getDocument(new IdRef(reply.getId()));
             return session.getDocument(new IdRef(replyDocumentModel.getId()));
         }, COMMENT_ADDED, DOCUMENT_CREATED);
     }
 
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.comment:OSGI-INF/notification-subscription-contrib.xml")
+    public void testAutoSubscribingOnlyOnceToNewComments() {
+        try (CapturingEventListener listener = new CapturingEventListener(COMMENT_ADDED)) {
```

Comment:
i think it will better if we work together to rework this test for more readability / simplicity

Reply:
As long as it reads easily I'm ok with it as it is.
Still open to suggestions though

Reply:
I think the idea is to test the auto-subscription behaviour only not the whole process i mean we have already which covers the sent emails/events. 
auto-subscription behaviour only -> create the needed users then ensure that that users haven't a subscription on  any comment event then create some comments / annotations and then ensure that every user had what is expected on term of subscription only by calling the different services on `NotificationManager` and then removes some of them and create another comments....and so on, i think it will simplify the test the final purpose to ensure that we have the subscription or not.

Reply:
The test has been simplified now, only tests autosubscription works and once unsubscribed no autoresub is done.